src/server.cpp

```diff
@@ -2712,19 +2636,17 @@ void Server::sendRequestedMedia(session_t peer_id,
 	}
 }
 
-void Server::sendDetachedInventory(const std::string &name, session_t peer_id)
+void Server::sendDetachedInventory(Inventory* inventory, const std::string &name, session_t peer_id)
 {
-	const auto &inv_it = m_detached_inventories.find(name);
-	const auto &player_it = m_detached_inventories_player.find(name);
+	std::string attachedPlayerName;
+	m_inventory_mgr->getAttachedPlayer(name, attachedPlayerName);
```

Comment:
I think it would be saner to use the inventory name as key, and call a function within `m_inventory_mgr` to retrieve `Inventory *` and the player name both at once. This practice is confusing.

Reply:
the problem i wanted to avoid is the double lookup on Inventory objects. I may try to refactor a little bit more to find a better solution

Reply:
when you have many detached inventories, and a player connects, you will lookup many times on the inventories to verify if they are attached. Maybe we should refactor detached inventories to have a ownership directly on it, then ?

Reply:
okay i found a way to solve it smoothly and drop the old lookup hack :)